### PR TITLE
Fixing conda

### PR DESCRIPTION
**Pull request by pjh5**

### CODE REVIEW

**Review by Yangqing — APPROVED** [review]

**Comment by pjh5** [comment]
This change will now encompass fixes needed for the conda_full build (which will no longer run in CI, as it must work in a special environment that can not be emulated there).

The conda-tests are still not working, due to opencv requiring specific protobuf versions that are hard to find without introducing gcc ABI issues.

**Comment by pjh5** [comment]
Looks like this is working for conda3, but conda2 still failing with a segfault during conda install P59201060

**Comment by pjh5** [comment]
\o/ conda2 works too! @caffe2bot test this please

File: .jenkins/build.sh
Comment (pjh5) [inline .jenkins/build.sh:57]: This was needed to fix an exception thrown by 'click' (used by onnx) that complained about detection incorrect locale configuration on the current machine. These same lines appear https://github.com/caffe2/caffe2/blob/master/docker/jenkins/common/install_anaconda.sh#L6 but I guess the values don't persist.

File: caffe2/core/predictor.h
Comment (pjh5) [inline caffe2/core/predictor.h]: see comment in caffe2/proto/caffe2.proto

File: .jenkins/test.sh
Comment (pjh5) [inline .jenkins/test.sh:97]: These changes needed to work with conda's different prefix in the site-packages name

File: caffe2/python/predictor/predictor_exporter_test.py
Comment (pjh5) [inline caffe2/python/predictor/predictor_exporter_test.py]: These tests both fail when the messages in metanet_pb2 are in their own file that imports caffe2_pb2

File: conda/cuda_full/build.sh
Comment (pjh5) [inline conda/cuda_full/build.sh:63]: This file is custom for the GPU cluster environment. It is not run in CI and will not work there, nor on "normal" setups. This was a special ask

File: scripts/read_conda_versions.sh
Comment (pjh5) [inline scripts/read_conda_versions.sh:181]: This is the script I used to scrape all the conda packages for mentions of cxx11 to guess which gcc ABI they had. It is not called by anything.

**Comment by pjh5** [comment]
This review fixes the conda tests for CI. Before now, conda builds were only tested by running 'from caffe2.python import core' and seeing that it worked. This enables all c++ and python tests to work on the conda builds too. After this lands, some conda builds will be moved from build-only to build-and-test in CI

**Comment by pjh5** [comment]
@Yangqing just an FYI this has changed significantly since you reviewed it
@salexspb this moves the metanet_pb2 into caffe2_pb2 b/c an error mentioned in a comment in that file. This change (and the associated renaming metanet_pb2.MetaNetDef() -> caffe2_pb2.MetaNetDef()) is the only change in the diff that will affect non-conda-build code

**Comment by pjh5** [comment]
Okay actually it still breaks on conda3 because of locale issues. But I'd like to check this in anyways to unblock conda2 tests

**Pull request merged**
